sql/core/src/main/scala/org/apache/spark/sql/catalyst/SQLBuilder.scala

```diff
@@ -356,8 +372,14 @@ class SQLBuilder(logicalPlan: LogicalPlan) extends Logging {
       }
     }
 
+    val broadcastHint = project match {
+      case p @ Project(projectList, Hint("BROADCAST", tables, child)) =>
+        if (tables.nonEmpty) s"/*+ MAPJOIN(${tables.mkString(", ")}) */" else ""
+      case _ => ""
+    }
     build(
       "SELECT",
+      broadcastHint,
```

Comment:
Hmm. For this one, I found more reasonable solution. Instead of adding new dummy `Project`, we can utilize the existing dummy `Project` on the `SQLTable`.

Basically, the root cause of this problem of `groupingSetToSQL` is that it removes the project by doing this `toSQL(project.child)`.